lib/mongoid/matcher/mod.rb

```diff
@@ -0,0 +1,22 @@
+module Mongoid
+  module Matcher
+
+    # @api private
+    module Mod
+      module_function def matches?(exists, value, condition)
+        unless Array === condition
+          raise Errors::InvalidQuery, "Unknown $mod argument #{condition}"
+        end
+        if condition.length() != 2
+          raise Errors::InvalidQuery, "BadValue malformed mod, invalid number of elements"
+        end
+        case condition
+        when Array
```

Comment:
This check isn't needed as it was already done earlier, you can remove the case and just leave line 15.